tests/test_client_session.py

```diff
@@ -271,6 +271,18 @@ def test_http_DELETE(session: Any, params: Any) -> None:
     ]
 
 
+async def test_http_empty_data_text(session: Any) -> None:
+    with mock.patch.object(session, "_connector", autospec=True) as connector_mock:
+        connector_mock.closed = False
+
+        # ClientOSError occurs after req has been produced, so we can just give up there.
+        with pytest.raises(aiohttp.client_exceptions.ClientOSError):
+            await session.post("https://foo.example", data="")
+
+        req = connector_mock.connect.await_args.args[0]
+        assert req.headers["Content-Type"] == "text/plain; charset=utf-8"
+
```

Comment:
Maybe there's a cleaner way to test this? I didn't see an obvious way, but this is a little messy with the mock causing that exception and a couple of warnings.